WriteVHDLSyntax.py

```diff
@@ -1533,15 +1644,21 @@ def writeProcMemoryLHSPorts(argname,mem,split = False):
         string_mem_ports += "      "+argname+"_dataarray_data_V_ce0       => open,\n"
         string_mem_ports += "      "+argname+"_dataarray_data_V_we0       => "
         string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_wea,\n"
-        string_mem_ports += "      "+argname+"_dataarray_data_V_address0  => "
-        string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_writeaddr,\n"
+        if "TPAR" in mem.inst and split == 1 : #FIXME TPAR addr from TP should be 10 bits
+            string_mem_ports += "      "+argname+"_dataarray_data_V_address0(9 downto 0)  => "
+            string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_writeaddr,\n"
+            string_mem_ports += "      "+argname+"_dataarray_data_V_address0(11 downto 10)  => "
+            string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_dummy,\n"
```

Comment:
@jasonfan393 Could we remove the hard-coded bit widths by, e.g., concatenating the inputs to the address port?:
```suggestion
            string_mem_ports += "      "+argname+"_dataarray_data_V_address0  => ("
            string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_dummy, "
            string_mem_ports += mem.mtype_short() + "_" + mem.var()+"_writeaddr),\n"
```